File: Project Euler/Problem 2 - Even Fibonacci numbers/Answers/Answer.kt
fun main() {
    print("Enter a number: ")
    val input: Int = readLine()!!.toInt()
    val res: Iterator<Int> = fib(input)
    var summation: Int = 0
    for (value in res) {
        if (value % 2 == 0) {
            summation += value
        }
    }
    println(summation)
}

fun fib(limit: Int) = iterator {
    var (a: Int, b: Int) = Pair(1, 2)
    while (a < limit) {
        a = b.also { b = a+b }
        yield (a)
    }
}

File: Project Euler/Problem 2 - Even Fibonacci numbers/Answers/Answer.js
function* fib (limit) {
    [a, b] = [1, 2]
    while (a < limit) {
        [a, b] = [b, a+b]
        yield a
    }
}

function mainModule(upper_bound) {
    summation = 0
    for (let value of fib(upper_bound)) {
        if (value % 2 == 0) {
            summation += value
        }
    }
    return summation
}

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout,
});

readline.question(`Enter a number: `, value => {
    console.log(mainModule(+value))
    readline.close();
});